### PR TITLE
Basic nuget package dependency support.

### PR DESCRIPTION
**Pull request by zaphar**
I've tested in a small workspace and was able to build a binary with a nuget package dependency.

It's probably still rough around the edges a bit but it's a start.

I couldn't find the gerrit repo to push to for code review so feel free to tell me where I should be putting this instead.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,14 +32,22 @@ load(
...
     "csharp_repositories",
     "csharp_configure",
+    "nuget_package",
 )
 csharp_repositories()
 csharp_configure()
+
+nuget_package(
+  name="some_name",
+  pkg_name="Some.Package",
```
Comment (davidzchen) [inline README.md]: Would it be better if we name the attributes `package` and `version` rather than `pkg_name` and `pkg_version`?

File: README.md [hunk 2]
```diff
@@ -32,14 +32,22 @@ load(
...
+  pkg_version="0.1.2",
```
Comment (davidzchen) [inline README.md]: Nit (buildifier style): add spaces before and after `=`

File: dotnet/csharp.bzl [hunk 3]
```diff
@@ -407,6 +417,104 @@ def csharp_configure():
...
+  inputs = set(ctx.files.srcs)
+  return struct(
+    name = ctx.label.name,
+    target_type = ctx.attr._target_type,
+    out = inputs,
+    dlls = inputs,
+    transitive_dlls = set([]),
+  )
+
+nuget_dep = rule(
```
Comment (davidzchen) [inline dotnet/csharp.bzl]: This seems similar to the [java_import](http://bazel.io/docs/be/java.html#java_import) rule. Do you think this rule might be useful for use cases other than nuget packages?

File: dotnet/csharp.bzl [hunk 4]
```diff
@@ -407,6 +417,104 @@ def csharp_configure():
...
+tell it where your nuget.exe is located. You may want to manage that binary
+in your repository as a result.
+
+Args:
+  nuget_package_source: list of sources to use for nuget package feeds.
+  pkg_name: name of the nuget package.
+  pkg_version: version of the nuget package (e.g. 0.1.2)
+"""
+
+NUNIT_BUILD_FILE = """
```
Comment (davidzchen) [inline dotnet/csharp.bzl]: Is this change intentional? I moved the BUILD file for nunit into a standalone BUILD file in f30f2f80984cea9cb53aa46ceacbc7c4046c2f89.

**Comment by davidzchen** [comment]
Awesome! Thanks for working on this!

It looks good for the most part. I have added some comments to your patch.

Also, can you rebase and squash your commits?

**Comment by damienmg** [comment]
(note: a new feature on github: you can squash commits when merging a PR,
you can even force to be the default \o/)

File: README.md (hunk 1 above)
Comment (zaphar) [inline README.md]: I'm not sure what the prevailing style is here. I have no strong preferences myself. Does the bazel community have a style for this sort of name?

File: dotnet/csharp.bzl (hunk 4 above)
Comment (zaphar) [inline dotnet/csharp.bzl]: Without it my csharp_test outside of this repository wouldn't build claiming it couldn't find the nunit.BUILD file. If there is a better way to fix that error I'm all for it.

**Comment by zaphar** [comment]
Sure no problem, I'll squash the commits once I've satisfied all the comments I like to save that operation for last in case I need to rollback single changes or something.

File: dotnet/csharp.bzl (hunk 3 above)
Comment (zaphar) [inline dotnet/csharp.bzl]: It certainly might. I haven't looked at the java_import rule. It's basically just a way to reference other dlls. perhaps it should just be called dll_import or dotnet_import?

It's also possible that some refactoring of the csharp_* rules could make this thin wrapper unnecessary and we could just use filegroups. This was the minimal I could do to make it work.

**Comment by zaphar** [comment]
Okay renamed pkg_name, pkg_version and nuget_package_source and fixed the style nits.

Let me know if you want me to remove the NUNIT_BUILD_FILE piece before I squash things down. It's not really related specifically to this change it just made it a bit easier to test some things.

**Comment by davidzchen** [comment]
Great, thanks!

Just wondering, did you run into any issues with having the nunit build file in a separate BUILD file while you were testing your change? If so, then I would be fine with keeping the NUNIT_BUILD_FILE change.

AFAIU, the `build_file_content` attribute was originally added as a workaround for some of the issues that we had with getting the rules to refer to the correct workspace (see bazelbuild/bazel#1075) but now that the issue is fixed and the rules have been migrated into separate repositories, we can switch back to using `build_file` instead.

**Comment by zaphar** [comment]
I decided to remove the build file change to keep the change targeted better. The fix can be implemented in a separate change. I think it's because the build file is referenced relatively not with a fully qualified name in the external repository.

If I got my git commands correct then this should now be squashed with the nunit build file stuff removed.

**Comment by davidzchen** [comment]
LGTM

Awesome. Thanks!

For the build_file fix, I think the proper fix is to use a label for `build_file` but let's do that in a separate change as you suggested.

**Comment by kchodorow** [comment]
This is broken, it re-defines csharp_configure a second time (see https://github.com/bazelbuild/rules_dotnet/blob/master/dotnet/csharp.bzl#L412 and https://github.com/bazelbuild/rules_dotnet/blob/master/dotnet/csharp.bzl#L503).  Please fix or roll back.

**Comment by davidzchen** [comment]
Sorry, missed this. Opened #8 to fix.
